vllm/model_executor/models/minicpm3.py

```diff
@@ -304,22 +181,22 @@ def forward(
         return output
 
 
-class MiniCPMDecoderLayer(nn.Module):
+class MiniCPM3DecoderLayer(MiniCPMDecoderLayer):
 
     def __init__(
         self,
         config,
         cache_config: Optional[CacheConfig] = None,
         quant_config: Optional[QuantizationConfig] = None,
     ) -> None:
-        super().__init__()
+        nn.Module.__init__(self)
```

Comment:
This is a bit hacky. Perhaps it would be better to have a base class with the common methods and have both MiniCPM and MiniCPM-3 inherit from it.

Reply:
Thank you for your advice. I have optimized the structure of my implementation. Please review and let me known if you have any further questions.

Reply:
It looks better now, thanks!